plugins/module_utils/networkfirewall.py

```diff
@@ -164,6 +180,73 @@ def _wait_rule_group_active(self, **params):
         waiter.wait(**params)
 
 
+class NFPolicyBoto3Mixin(NetworkFirewallBoto3Mixin):
+    # Paginators can't be (easily) wrapped, so we wrap this method with the
+    # retry - retries the full fetch, but better than simply giving up.
+    @AWSRetry.jittered_backoff()
+    def _paginated_list_policies(self, **params):
+        paginator = self.client.get_paginator('list_firewall_policies')
+        result = paginator.paginate(**params).build_full_result()
+        return result.get('FirewallPolicies', None)
+
+    @Boto3Mixin.aws_error_handler('list all firewall policies')
+    def _list_policies(self, **params):
+        return self._paginated_list_policies(**params)
+
+    @Boto3Mixin.aws_error_handler('describe firewall policy')
+    def _describe_policy(self, **params):
+        try:
+            result = self.client.describe_firewall_policy(aws_retry=True, **params)
+        except is_boto3_error_code('ResourceNotFoundException'):
+            return None
+
+        update_token = result.get('UpdateToken', None)
+        if update_token:
+            self._update_token = update_token
+        policy = result.get('FirewallPolicy', None)
+        metadata = result.get('FirewallPolicyResponse', None)
+        return dict(FirewallPolicy=policy, FirewallPolicyMetadata=metadata)
+
+    @Boto3Mixin.aws_error_handler('create firewall policy')
+    def _create_policy(self, **params):
+        result = self.client.create_firewall_policy(aws_retry=True, **params)
```

Comment:
wrap into try/except

Reply:
`@Boto3Mixin.aws_error_handler` is doing this.

Reply:
ah. ok